app/web/features/auth/notifications/NotificationSettings.tsx

```diff
@@ -0,0 +1,87 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import { notificationSettingsQueryKey } from "features/queryKeys";
+import { Error as GrpcError } from "grpc-web";
+import {
+  GetNotificationSettingsRes,
+  SetNotificationSettingsRes,
+} from "proto/notifications_pb";
+import { useTranslation } from "react-i18next";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { service } from "service";
+
+interface NotificationSettingFormData {
+  newNotificationsEnabled: boolean;
+}
+
+export default function NotificationSettings({
+  className,
+}: {
+  className: string;
+}) {
+  const { t } = useTranslation("auth");
+
+  const queryClient = useQueryClient();
+
+  const { data, error, isLoading } = useQuery<
+    GetNotificationSettingsRes.AsObject,
+    GrpcError
+  >(
+    notificationSettingsQueryKey,
+    service.notifications.getNotificationSettings
+  );
+
+  const mutation = useMutation<
+    SetNotificationSettingsRes.AsObject,
+    GrpcError,
+    NotificationSettingFormData
+  >(
+    ({ newNotificationsEnabled }) =>
+      service.notifications.setNotificationSettings(newNotificationsEnabled),
```

Comment:
```suggestion
    async ({ newNotificationsEnabled }) => {
      if (newNotificationsEnabed === undefined) throw Error("Current setting not loaded yet.");
      return service.notifications.setNotificationSettings(newNotificationsEnabled);
    },
```
This kind of thing would be most "correct", as it would display the thrown error message in the error alert, but maybe it's a bit pedantic. See below for alternative. (If you went with this you'd have to make the type `newNotificationsEnabled?: boolean;`